src/care/KeyValueSorter.h

```diff
@@ -223,6 +223,24 @@ class KeyValueSorter<T, RAJADeviceExec> {
          setFromArray(len, arr);
       }
 
+      ///////////////////////////////////////////////////////////////////////////
+      /// @author Alan Dayton
+      /// @brief Constructor
+      /// Allocates space and initializes the KeyValueSorter by copying
+      ///    elements and ordering from the given managed array
+      /// @param[in] len - The number of elements to allocate space for
+      /// @param[in] arr - The managed array to copy elements from
+      /// @return a KeyValueSorter instance
+      ///////////////////////////////////////////////////////////////////////////
+      KeyValueSorter<T, RAJADeviceExec>(const size_t len, const host_device_ptr<const T>& arr)
+      : m_len(len)
+      , m_ownsPointers(true)
+      , m_keys(len, "m_keys")
+      , m_values(len, "m_values")
+      {
+         setFromArray(len, arr.data());
```

Comment:
There is an overload of setFromArray that takes a `const host_device_ptr<T>&` as an argument. But the method never writes to it, so you should be able to make the argument a `const host_device_ptr<const T>&` and not have to use `.data()` here since that results in an extra copy to the host.

Reply:
If the compiler doesn't like `const T` when `T` is already const (i.e. if `T` is `const real8`, I don't know how it would handle `const const real8`), then you could use type traits: `using T_const = typename std::add_const<T>::type;` Then you should be able to use `const host_device_ptr<T_const>&`. You'd have to include the <type_traits> header in this case.

Reply:
const host_device_ptr<const T>& worked fine, changed setFromArray to be consistent